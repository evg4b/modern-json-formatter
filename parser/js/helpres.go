package js

func errorResponse(error string) map[string]any {
	return map[string]any{
		"type":  "error",
		"error": error,
	}
}

func successResponse(value map[string]any) map[string]any {
	return map[string]any{
		"type":  "response",
		"value": value,
	}
}
