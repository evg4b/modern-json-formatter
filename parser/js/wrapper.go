package js

import (
	"github.com/evg4b/modern-json-formatter/parser/decode"
	"syscall/js"
)

func ParseWrapper() js.Func {
	return js.FuncOf(func(_ js.Value, args []js.Value) any {
		if len(args) != 1 {
			return map[string]interface{}{
				"type":  "error",
				"error": "Invalid arguments passed",
			}
		}

		jsonTree, err := decode.Decode(args[0].String())
		if err != nil {
			return map[string]any{
				"type":  "error",
				"error": err.Error(),
			}
		}

		return jsonTree
	})
}
