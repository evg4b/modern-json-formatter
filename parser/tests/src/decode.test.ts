import '../../wasm_exec.js';
import {loadWasm} from "./load-wasm";


declare function parseJSON(data: string): ParserResponse;

describe('sum module', () => {
    loadWasm();

    test('should be defined', () => {
        expect(parseJSON).toBeDefined();
    });

    test('sum', () => {
        const d = parseJSON('{ "a": 1, "b": 2 }');
        expect(d).toMatchSnapshot()
    });
});
