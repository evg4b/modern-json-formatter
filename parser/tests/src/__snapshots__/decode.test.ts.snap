// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sum module sum 1`] = `
{
  "type": "response",
  "value": {
    "properties": [
      {
        "key": "a",
        "value": {
          "type": "number",
          "value": "1",
        },
      },
      {
        "key": "b",
        "value": {
          "type": "number",
          "value": "2",
        },
      },
    ],
    "type": "object",
  },
}
`;
